app/main/posts/modify/post-video.directive.js

```diff
@@ -0,0 +1,92 @@
+module.exports = PostVideo;
+
+PostVideo.$inject = [];
+
+function PostVideo() {
+    return {
+        restrict: 'E',
+        scope: {
+            videoUrl: '='
+        },
+        templateUrl: 'templates/main/posts/modify/video.html',
+        controller: PostVideoController
+    };
+}
+
+PostVideoController.$inject = [
+    '$scope',
+    '$sce',
+    'Util'
+];
+
+function PostVideoController(
+    $scope,
+    $sce,
+    Util
+) {
+    $scope.constructIframe = constructIframe;
+    // $scope.getVideoThumbnail = getVideoThumbnail;
+    activate();
+
+    function activate() {
+        // Is this dody? I dunno it seems dodgy
+        $scope.videoUrl = $sce.trustAsResourceUrl($scope.videoUrl);
```

Comment:
So yeah, this is the real problem here, we have to trust the url. I will add a regex to check that it is a video url based on what we think a video url is but this is still potentially going to cause problems. 

We are assuming that the url in the iframe is going to be kosher. It would not be hard to trick our regex into allowing a url that looked right but was malicious. I'm not positive what to do about this. 

For the moment, I have prevented them from entering urls that don't match the format of youtube or vimeo urls. The fact they are gameable urls is sort of difficult to fix because we don't control vimeo or youtube and as such can't force their url format super strictly.

So if someone is able to insert a malicious url that passes our regex and loads in the iframe it could infect the machine of someone viewing the page. 

I'll think about more about sanitisation.